File: day1/main.c
#include <stdio.h>
#include <fcntl.h>

void p1() {
	int floor;
	char c;
	int fd = open("input.txt", O_RDONLY);
	if (fd == -1) {
		printf("Couldn't open the file\n");
		exit(1);
	}
	while (read(fd, &c, 1)) floor += (c == '(') + -(c == ')');
	ssize_t s = read();
	close(fd);
	printf("%d\n", floor);
}

void p2() {
	int floor = 0;
	char c;
	int n = 0;
	int fd = open("input.txt", O_RDONLY);
	if (fd == -1) {
		printf("Couldn't open the file\n");
		exit(1);
	}
	while (read(fd, &c, 1)) {
		if (c == '(') floor++;
		else floor--;
		n++;
		if (floor < 0) break;
	}
	ssize_t s = read();
	close(fd);
	printf("%d\n", n);
}

int main(void) {
	p1();
	p2();
	exit(0);
}

File: day1/main.f90
PROGRAM main
	IMPLICIT NONE
	INTEGER :: io, level, i
	CHARACTER(len = 7000) :: str

	PRINT *, "Part one:"
	OPEN(newunit=io, file="input.txt", status="old", action="read")
	READ(io, *) str
	DO i = 0, 7000
		IF (str(i:i) .eq. '(') THEN
			level = level + 1
		ELSE IF (str(i:i) .eq. ')') THEN
			level = level - 1
		ENDIF
	ENDDO
	PRINT *, level
	CLOSE(io)
	level = 0

	PRINT *, "Part two:"
	OPEN(newunit=io, file="input.txt", status="old", action="read")
	READ(io, *) str
	DO i = 0, 7000
		IF (str(i:i) .eq. '(') THEN
			level = level + 1
		ELSE IF (str(i:i) .eq. ')') THEN
			level = level - 1
		ENDIF
		IF (level < 0) THEN
			level = i
			EXIT
		ENDIF
	ENDDO
	PRINT *, level
	CLOSE(io)

END PROGRAM main
